packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -96,91 +116,49 @@ export class SemanticViewerResultsRunner extends ResultsRunner {
         pivotLayoutOptions: VizPivotLayoutOptions[];
     } {
         return {
-            indexLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.DIMENSION) {
-                    acc.push({
-                        reference: column.reference,
-                        type: this.getAxisType(column),
-                    });
-                }
-                return acc;
-            }, [] as VizIndexLayoutOptions[]),
-            valuesLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.METRIC) {
-                    acc.push({
-                        reference: column.reference,
-                    });
-                }
-                return acc;
-            }, [] as VizValuesLayoutOptions[]),
-            pivotLayoutOptions: this.columns.filter((column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                return field?.kind === FieldType.DIMENSION;
-            }),
+            indexLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
+            valuesLayoutOptions: this.metrics.map((metric) => ({
+                reference: metric.name,
+            })),
+            pivotLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
         };
     }
 
     defaultPivotChartLayout(): VizChartLayout | undefined {
-        const xColumn = this.columns.find((column) => {
-            const field =
-                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                    this.fields,
-                    column.reference,
-                );
-            return field?.kind === FieldType.DIMENSION;
-        });
+        const xColumn = this.dimensions[0];
+        const yColumn = this.metrics[0];
```

Comment:
Something similar here, were setting the `default` config axis to `dimensions` and/or `metrics` that are not included in the results (e.g. here I selected `type` dimension and `orders` metric but the default config was `name` and `orders total` which are the first in the lists)

![image](https://github.com/user-attachments/assets/9ba0d3bb-3b19-4174-9dd3-b0d985d40719)

Reply:
I believe this is fixed.